### PR TITLE
Electron app to show git repository contributors by directory

### PR DESCRIPTION
**Pull request by ouchadam**
Electron desktop app (nodejs) to attempt to show the ownership of all the directories of a project

screen shot 2016-05-01 at 13 26 19

Usage - 

Must have `git` on the path and a writable `/tmp/` directory (which should be everyone except windows peeps).

clone the repo, cd into `electron-contrib` and run `npm install && npm start`

specify the full path to the git repository to analyse and submit

### CODE REVIEW

File: electron-contrib/contrib.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+      }
+      executeShortLog(gitRepoPath, file, function(result) {
+        callback(result);
+        checkInt(gitRepoPath, file, callback);
+      });
+  });
+}
+
+function executeShortLog(gitRepoPath, file, callback) {
+  exec(shortlogCommand(gitRepoPath, file), { cwd : "/tmp/" }, function(err, stdout, stderr) {
```
Comment (frapontillo) [inline electron-contrib/contrib.js]: We don't need no double quotes (cit.)

**Comment by frapontillo** [comment]
Would you like to give a chance to [nodegit](https://github.com/nodegit/nodegit)? @danybony and I were also interested in using it for the AOSP changelog project. Should also be faster than spawning and reading stdout.

File: electron-contrib/renderer.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  while (resultElement.children.length > 0) {
+    root.appendChild(resultElement.children[0]);
+  }
+}
+
+waitForWindow(onReady);
+
+function waitForWindow(callback) {
+  if (document.readyState === 'interactive' || document.readyState === 'complete') {
+      callback();
```
Comment (xrigau) [inline electron-contrib/renderer.js]: dis indentation tho

**Comment by blundell** [comment]
add a README.md 🙈

**Comment by ouchadam** [comment]
@frapontillo I had a quick look at `nodegit` but couldn't find any api for `shortlog` :cry:

File: electron-contrib/renderer.js (hunk 2 above)
Comment (ouchadam) [inline electron-contrib/renderer.js]: atom auto format has a mind of its own

File: electron-contrib/contrib.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
+const fs = require('fs');
+const exec = require('child_process').exec;
+
+module.exports = {
+  check: function(directory, callback) {
```
Comment (frapontillo) [inline electron-contrib/contrib.js]: Or `check: check` 😁

Comment (ouchadam) [inline electron-contrib/contrib.js]: I knew it! will change

**Comment by xrigau** [comment]
@ouchadam add a README and I'll :shipit:

**Comment by blundell** [comment]
👍

**Pull request merged**